### PR TITLE
feat(nuxt,schema): `pages:resolved` hook + scan meta post extend

### LINKED ISSUE #27814: `layout: false` does not remove the layout when multiple routes use the same page component
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v22.1.0
- Nuxt Version:     3.12.0
- CLI Version:      3.12.0
- Nitro Version:    2.9.6
- Package Manager:  npm@10.7.0
- Builder:          -
- User Config:      devtools, runtimeConfig, vite, alias, modules, image, hooks, nitro, serverMiddleware, build, css
- Runtime Modules:  @pinia/nuxt@0.5.1, @tixel/ui-nuxt@0.0.7, @tixel/component-library@0.4.0, nuxt-jsonld@2.0.8, @pinia-plugin-persistedstate/nuxt@1.2.0, @nuxt/devtools@1.3.6, @nuxt/image@1.0.0-rc.3, @vueuse/nuxt@10.11.0, @nuxtjs/device@3.1.1, @nuxt/content@2.12.1
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-m1cmbw?file=pages%2Fexample-page.vue

### Describe the bug

On the reproduction URL

Go to `/example-page`

You will see that there is no default layout set, because the page component has:

```
definePageMeta({
  layout: false,
});
```

However, now go to `/example-page-2` and you will see that this page still includes the layout, despite using the same component as the other route.

Both pages should not render the layout.

See the routes are rendered in `routes.ts`

### Additional context

This only occurs in `v3.12.x`, rolling back to `v3.11.x` fixes the issue.

### Logs

_No response_

### LINKED ISSUE #28122: Nuxt 3.12.0 no longer calls route key functions in definePageMeta for routes created by the pages:extend hook
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.12.1`
- Nuxt Version:     `3.12.3`
- CLI Version:      `3.12.0`
- Nitro Version:    `2.9.7`
- Package Manager:  `npm@10.5.0`
- Builder:          `-`
- User Config:      `app`, `appConfig`, `vite`, `css`, `components`, `modules`, `formkit`, `experimental`, `nitro`, `sourcemap`, `runtimeConfig`, `postcss`, `devtools`, `typescript`
- Runtime Modules:  `nuxt-vitest@0.10.5`, `@vueuse/nuxt@10.11.0`, `@pinia/nuxt@0.5.1`, `@formkit/nuxt@1.6.5`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-c2dzdp

Instructions are on the site.

### Describe the bug

Starting with Nuxt 3.12.0, Nuxt is no longer calling key functions defined in `definePageMeta` for pages added via `pages:extend` hook.  This results in pages re-mounting unexpectedly.  In our case, we have a couple of routes all going to the same page component.  We use `definePageMeta/key` to make sure the app re-mounts when specific url data changes.

It seems to be related to this change, https://github.com/nuxt/nuxt/pull/27134/files.
If I set `scanPageMeta: false`, it works as expected.

### Additional context
Possibly related to https://github.com/nuxt/nuxt/issues/28086

Pages added via /pages folder seem fine.

### PR DESCRIPTION
**Pull request by BobbieGoede**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/28122
resolves https://github.com/nuxt/nuxt/issues/27814

### 📚 Description
This is a proof of concept for an additional hook meant to modify pages, while keeping `pages:extends` for adding pages.

### CODE REVIEW

File: packages/nuxt/src/pages/utils.ts [hunk 1]
```diff
@@ -64,17 +64,14 @@ export async function resolvePagesRoutes (): Promise<NuxtPage[]> {
...
   if (shouldAugment) {
-    const augmentedPages = await augmentPages(pages, nuxt.vfs)
-    await nuxt.callHook('pages:extend', pages)
-    await augmentPages(pages, nuxt.vfs, augmentedPages)
-    augmentedPages.clear()
-  } else {
-    await nuxt.callHook('pages:extend', pages)
+    await augmentPages(pages, nuxt.vfs)
+    await nuxt.callHook('pages:resolved', pages)
   }
```
Comment (huang-julien) [inline packages/nuxt/src/pages/utils.ts:75]:
```suggestion
  }
  await nuxt.callHook('pages:resolved', pages)
```

maybe right before returning pages ?

**Comment by BobbieGoede** [comment]
These changes look good! I think this should provide the flexibility we're currently missing to use the augmented pages in `pages:resolved`.

@huang-julien 
Can you check if this works in your project with `pnpm add https://pkg.pr.new/BobbieGoede/i18n/@nuxtjs/i18n@57` (preview release of https://github.com/BobbieGoede/i18n/pull/57, run with `EXPERIMENTAL_HOOK=1`), let me know if other changes need to be made.

**Review by huang-julien — COMMENTED** [review]
Looks great ! Thanks to both of you :heart: 

I'll take a look and test ASAP :eyes:

**Comment by danielroe** [comment]
@huang-julien merging to iterate on this (as it's still marked experimental) - let me know if you have any concerns 🙏

**Pull request merged**